rewrite-java/src/main/java/org/openrewrite/java/tree/JContainer.java

```diff
@@ -73,7 +73,7 @@ private JContainer(Space before, List<JRightPadded<T>> elem, Markers markers) {
     }
 
     public JContainer<T> withElem(List<JRightPadded<T>> elem) {
-        return build(getBefore(), elem, markers);
+        return this.elem == elem ? this : build(getBefore(), elem, markers);
```

Comment:
Adds idempotence to JContainer.withElem. This is necessary because JavaVisitor.visitContainer provides idempotence, but also requires that the J in JContainer<J> implement acceptJava in order for any spaces contained in the elem to be visited. If not, TreeVisitor.visit is called which is just an identity implementation. This is the case for J.Try.Resource. To deal with this, we must use a withElem(withPrefix(visitSpace))) type of chain to visit the space in the elem, but withElem needs to be idempotent. You might need to read that a few dozen times.

This is a less invasive fix because adding acceptJava to J.Try.Resource requires us to add a visitor method to JavaVisitor, when (I think/hope) that level of involvement isn't really necessary.